src/app/product.service.ts

```diff
@@ -0,0 +1,60 @@
+import { Injectable } from '@angular/core';
+import { Observable, of } from 'rxjs';
+import { catchError, map } from 'rxjs/operators';
+import { HttpClient } from '@angular/common/http';
+
+import { Product } from './product';
+import { SoldProducts } from './product';
+
+
+@Injectable({
+  providedIn: 'root'
+})
+export class ProductService {
+  private productsUrl = 'https://private-anon-cfb1043af1-swoproducts.apiary-mock.com/soldProducts'; 
+
+  constructor(private http: HttpClient) { }
+
+  /** GET soldProducts from the server */
+  getSoldProducts(): Observable<SoldProducts> {
+    return this.http.get<SoldProducts>(this.productsUrl);
+  }
+
+
+  /** GET productsTargetValue from the server */
+  getSoldProductsTargetValue(): Observable<any> {
+    return this.http.get<SoldProducts>(this.productsUrl).pipe(
```

Comment:
You already have http.get in getSoldProducts(). Good idea is to share this call and result with all other methods. Or even to create a shareable stream as this is just a bind. Example:
productsData: Observable<SoldProducts> = return this.http.get<SoldProducts>(this.productsUrl).pipe(share());